### PR TITLE
Explicitly clear the CSR_done bit

### PR DESCRIPTION
**Pull request by shuoniu-intel**
Also, update the name of some CSR bit macros to match the terminology in the compiler KernelCRA.

Tested locally in qshell2, passed L2.

### CODE REVIEW

File: src/acl_kernel_if.cpp
Comment (pcolberg) [inline src/acl_kernel_if.cpp:47]: Since the versioning scheme changed from FPGA SDK to oneAPI anyway, how about using `_` inplace of `.`?
```suggestion
#define CSR_VERSION_ID_2022_2 (4)
```

File: src/acl_kernel_if.cpp
Comment (pcolberg) [inline src/acl_kernel_if.cpp:95]: Could you commit this as a separate change and motivate the renaming in the commit message?

Each runtime commit should ideally be atomic, i.e., a single, self-consistent change.

File: src/acl_kernel_if.cpp
Comment (pcolberg) [inline src/acl_kernel_if.cpp:1478]: The above will break when the CSR version is bumped once more.
```suggestion
      // Since csr version 19, done bit is cleared when finish counter is read.
      // Since csr version 2022.3, done bit needs to be cleared explicitly.
      if (kern->csr_version == CSR_VERSION_ID_181 ||
          kern->csr_version >= CSR_VERSION_ID_2022_3) {
```

File: src/acl_kernel_if.cpp
Comment (pcolberg) [inline src/acl_kernel_if.cpp:1143]: How about using a range for future compatibility?
```suggestion
  if (!(kern->csr_version >= CSR_VERSION_ID_181 &&
        kern->csr_version <= CSR_VERSION_ID)) {
```

File: src/acl_kernel_if.cpp
Comment (pcolberg) [inline src/acl_kernel_if.cpp:48]: We could introduce an alias for the latest version:
```suggestion
#define CSR_VERSION_ID_2022_3 (5)
#define CSR_VERSION_ID CSR_VERSION_ID_2022_3
```
The explicitly versioned constant may then be used below.

**Review by pcolberg — CHANGES_REQUESTED** [review]
Thanks @shuoniu-intel, looks good! I have made some minor suggestions to ease future bumps of the CSR version, as well as breaking up the rename of CSR bits into a separate commit.

File: src/acl_kernel_if.cpp
Comment (shuoniu-intel) [reply src/acl_kernel_if.cpp:95]: Sure. I'll remove this renaming change from this patch.

Comment (pcolberg) [reply src/acl_kernel_if.cpp:95]: Thanks @shuoniu-intel! You can still have the renaming in this PR, just a separate commit. We use the rebase-merge strategy for this repository, so your multiple commits will be preserved as is on merge into `main`.

Comment (shuoniu-intel) [reply src/acl_kernel_if.cpp:95]: Cool! Good to know that commits in a PR will not be combined after merging in this repository.

File: src/acl_kernel_if.cpp
Comment (pcolberg) [inline src/acl_kernel_if.cpp:47]: On second thought this should be named after the first release in which it was introduced, which is [19.1](https://www.intel.com/content/www/us/en/software-kit/664561/intel-fpga-sdk-for-opencl-pro-edition-software-version-19-1.html).
```suggestion
#define CSR_VERSION_ID_19_1 (4)
```

**Review by pcolberg — CHANGES_REQUESTED** [review]

**Comment by pcolberg** [comment]
Thanks @shuoniu-intel! I will merge once the integration testing has passed.

**Review by pcolberg — APPROVED** [review]

**Pull request merged**
